tests/core/operations/test_transform.py

```diff
@@ -459,14 +481,111 @@ def test_map_coordinate_systems_long_path(full_sdata):
+            if isinstance(element, (SpatialImage, MultiscaleSpatialImage)):
+                assert np.allclose(a, np.array([[1 / k, 0], [0, 1]]))
+            else:
+                assert np.allclose(a, np.array([[1 / k, -k / k], [0, 1]]))
+        else:
+            assert set(d.keys()) == {"my_space"}
+            if isinstance(element, (SpatialImage, MultiscaleSpatialImage)):
+                assert np.allclose(a, np.array([[1, k], [0, 1]]))
+            else:
+                assert np.allclose(a, np.array([[1, 0], [0, 1]]))
+
+    # this calls transform_element_to_coordinate_system() internally()
+    _ = full_sdata.transform_to_coordinate_system("my_space", maintain_positioning=maintain_positioning)
+
+
+@pytest.mark.parametrize("maintain_positioning", [True, False])
+def test_transform_elements_and_entire_spatial_data_object_multi_hop(
+    full_sdata: SpatialData, maintain_positioning: bool
+):
+    k = 10.0
+    scale = Scale([k], axes=("x",))
+    for element in full_sdata._gen_elements_values():
         set_transformation(element, scale, "my_space")
-        sdata.transform_element_to_coordinate_system(element, "my_space")
-    sdata.transform_to_coordinate_system("my_space")
+
+    # testing the scenario "element1 -> cs1 <- element2 -> cs2" and transforming element1 to cs2
+    translation = Translation([k], axes=("x",))
+    full_sdata["proxy_element"] = full_sdata.shapes["multipoly"].copy()
+    set_transformation(
+        full_sdata["proxy_element"], {"multi_hop_space": translation, "my_space": Identity()}, set_all=True
+    )
+
+    # otherwise we have multiple paths to go from my_space to multi_hop_space
+    for element in full_sdata._gen_elements_values():
+        d = get_transformation(element, get_all=True)
+        assert isinstance(d, dict)
+        if "global" in d:
+            remove_transformation(element, "global")
+
+    for element in full_sdata._gen_elements_values():
+        # from spatialdata.transformations.operations import _build_transformations_graph
+        #
+        # g = _build_transformations_graph(full_sdata)
+        # import networkx as nx
+        #
+        # nx.draw(g, with_labels=True)
+        # import matplotlib.pyplot as plt
+        #
+        # plt.show()
+        transformed_element = full_sdata.transform_element_to_coordinate_system(
+            element, "multi_hop_space", maintain_positioning=maintain_positioning
+        )
+        temp = SpatialData(
+            images=dict(full_sdata.images),
+            labels=dict(full_sdata.labels),
+            points=dict(full_sdata.points),
+            shapes=dict(full_sdata.shapes),
+            table=full_sdata.table,
+        )
+        temp["transformed_element"] = transformed_element
+        t = get_transformation_between_coordinate_systems(temp, temp["transformed_element"], "multi_hop_space")
+        a = t.to_affine_matrix(input_axes=("x",), output_axes=("x",))
+        d = get_transformation(transformed_element, get_all=True)
```

Comment:
Can you please use more informative variable names?

Reply:
I agree, here it looks ugly. In general often in the code I use `t` for transformation, `a` for affine and `d` for the dict of transformations returned by `get_transformation(..., get_all=True)` (which is followed by `assert isinstance(d, dict)` to address `mypy`. So I think the `d` letter is ok.